### DISCUSSION TITLE
How do we access the swagger UI?

### DISCUSSION THREAD

**Discussion by MartinDawson** [Green-Software-Foundation/carbon-aware-sdk#378] [General]
I have a swagger yaml here: https://carbon-aware-dev.azurewebsites.net/swagger/index.html

How do we host the swagger UI? I was under the assumption that it would be automatic after v1.1.0 release?

I.e like you have here: https://carbon-aware-api.azurewebsites.net/swagger/index.html

I have copied the same DockerFile as you guys have:

```Dockerfile
FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env
WORKDIR /app

RUN wget https://github.com/Green-Software-Foundation/carbon-aware-sdk/archive/refs/tags/v1.1.0.zip

RUN apt-get update && \
  apt-get install unzip && \
  unzip v1.1.0.zip && \
  rm v1.1.0.zip

# Use implicit restore to build and publish
RUN dotnet publish carbon-aware-sdk-1.1.0/src/CarbonAware.WebApi/src/CarbonAware.WebApi.csproj -c Release -o publish
# Generate OpenAPI spec
WORKDIR /app/carbon-aware-sdk-1.1.0/src/CarbonAware.WebApi/src
RUN dotnet tool restore && \
  mkdir -p /app/publish/wwwroot/api/v1 && \
  dotnet tool run swagger tofile --output /app/publish/wwwroot/api/v1/swagger.yaml --yaml /app/publish/CarbonAware.WebApi.dll v1

# Build runtime image
FROM mcr.microsoft.com/dotnet/aspnet:6.0
# Install curl for health check
RUN apt-get update && \
  apt-get install -y --no-install-recommends curl
# Copy artifacts from build-env
WORKDIR /app
COPY --from=build-env /app/publish .
ENTRYPOINT ["dotnet", "CarbonAware.WebApi.dll"]
```

Thanks

**Comment by MartinDawson** [comment]
Any ideas on this?

**Comment by MartinDawson** [comment]
Figured it out.

Must put `ENV ASPNETCORE_ENVIRONMENT=Development` in the development configuration on my azure app.

That then runs through this code:

```c#
if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();   
}
```

**Comment by YaSuenag** [comment]
In [WebAPI container](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pkgs/container/carbon-aware-sdk), you can access Swagger document with following URL: http://<webapi container>/api/v1/swagger.yaml

**Comment by YaSuenag** [comment]
FYI: I've sent PR for WebAPI demonstration on Podman with Swagger UI: https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/340
